gym/core.py

```diff
@@ -207,52 +207,19 @@ class Wrapper(Env):
     """
     def __init__(self, env):
         self.env = env
-        self.action_space = self.env.action_space
-        self.observation_space = self.env.observation_space
-        self.reward_range = self.env.reward_range
-        self.metadata = self.env.metadata
 
     def __getattr__(self, name):
         if name.startswith('_'):
             raise AttributeError("attempted to get missing private attribute '{}'".format(name))
         return getattr(self.env, name)
 
     @property
-    def spec(self):
-        return self.env.spec
-
-    @classmethod
-    def class_name(cls):
-        return cls.__name__
-
-    def step(self, action):
-        return self.env.step(action)
-
-    def reset(self, **kwargs):
-        return self.env.reset(**kwargs)
-
-    def render(self, mode='human', **kwargs):
-        return self.env.render(mode, **kwargs)
-
-    def close(self):
-        return self.env.close()
-
-    def seed(self, seed=None):
-        return self.env.seed(seed)
-
-    def compute_reward(self, achieved_goal, desired_goal, info):
-        return self.env.compute_reward(achieved_goal, desired_goal, info)
+    def unwrapped(self):
+        return self.env.unwrapped
 
     def __str__(self):
         return '<{}{}>'.format(type(self).__name__, self.env)
 
-    def __repr__(self):
```

Comment:
should we rather delete __str__ method? To my understanding. if __str__ is not specified, `str(x)` will call x.__repr__(), but if `__repr__` is not specified, `repr(x)` will *not* call `x.__str__()`

Reply:
@pzhokhov Thanks for pointing this out. You are right. Also according to the discussion [here](https://stackoverflow.com/a/2626364/993191), it is better to implement `__repr__` and let contains default behavior for `__str__ = __repr__`